.github/workflows/quality-assurance.yml

```diff
@@ -0,0 +1,86 @@
+name: quality-assurance
+
+on:
+  pull_request:
+
+env:
+  python-version: "3.9"
+  rust-version: "1.61.0"
+  node-version: "16"
+
+jobs:
+  check-quality-assurance:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b
```

Comment:
can you also add a comment to indicate which tag this commit is supposed to stand for (yeah the commit is safer but also less readable when we want to do an update ^^)
```
 - uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b  # tag foobar
```

Reply:
done